draft-ietf-tls-dtls13.md

```diff
@@ -1350,6 +1323,83 @@ ClientHello                 -------->
 ~~~~
 {: #dtls-msg-epoch title="Example DTLS Exchange with Epoch Information"}
 
+
+# ACK Message {#ack-msg}
+
+The ACK handshake message is used by an endpoint to indicate the
+TLS records it has received from the other side. ACK is not
+a handshake message but is rather a separate content type,
+with code point TBD. This avoids it consuming space in the
+handshake message sequence. Note that ACKs can still be
+piggybacked on the same UDP datagram as handshake records.
+
+~~~~
+struct {
+       uint64 record_numbers<8..2^16-1>;
+} ACK;
+~~~~
+
+record_numbers:
+
+: a list of the records in the current flight which the endpoint has
+received, in numerically increasing order. ACKs only cover the current
+outstanding flight (this is possible because DTLS is generally a
+lockstep protocol). Thus, an ACK from the server would not cover
+both the ClientHello and the client's Certificate. Implementations
+can accomplish this by clearing their ACK list upon receiving the
+start of the next flight.
+
+ACK messages are sent with the then-current sending key.
+[TODO: I think we need rules about how you have to reject ACKs
+in the wrong phase. Note that this means that ACKs for the client's last flight
+are sent with the application key. We could alter this rule if
+we wanted.]
+
+
+## Sending ACKs
+
+    When an implementation receives a partial flight, it SHOULD generate
+    an ACK that covers the messages from that flight which it has
+    received so far. Implementations have some discretion about when
+    to generate ACKs, but it is RECOMMENDED that they do so under
+    two circumstances: 
+
+    - When they receive a message or fragment which is out of order,
+    either because it is not the next expected message or because
+    it is not the next piece of the current message.
+
+    - When they have received part of a flight and do not immediately
+    receive the rest of the flight (which may be in the same UDP
+    datagram). A reasonable approach here is to
+    set a timer for 1/4 the current retransmit timer value when
```

Comment:
It's actually pretty much undefined. We mostly just say "start with a timer of X and then back off".  For the main handshake, generally the only flight that is sent with an RTT estimate is the client's second flight, so it's kind of ad hoc. @ianswett would welcome wording here.